lib/snowflex/transport/http.ex

```diff
@@ -0,0 +1,650 @@
+defmodule Snowflex.Transport.Http do
+  @moduledoc """
+  REST API transport implementation for Snowflake.
+  See: https://docs.snowflake.com/en/developer-guide/sql-api/reference
+
+  ## Configuration Options
+
+  The HTTP transport supports the following configuration options:
+
+  ### Required Options
+  * `:account_name` - Your Snowflake account identifier (e.g., "my-org-my-account")
+  * `:username` - Your Snowflake username
+  * `:private_key_path` - Path to your private key file (PEM format)
+  * `:public_key_fingerprint` - Fingerprint of your public key
+
+  ### Optional Options
+  * `:database` - Default database to use
+  * `:schema` - Default schema to use
+  * `:warehouse` - Default warehouse to use
+  * `:role` - Default role to use
+  * `:timeout` - Query timeout in milliseconds (default: 45 seconds)
+  * `:token_lifetime` - JWT token lifetime in milliseconds (default: 10 minutes)
+  * `:private_key_password` - Password for the private key (if encrypted)
```

Comment:
Would be nice to explicitly mention how to handle private keys without a password here.

Reply:
OK that's intuitive and no further documentation should be required. That was not the behaviour when I first tried out this PR, see my other comment.